### PR TITLE
Fixes #1794 residuals vs time plot not refreshing

### LINKED ISSUE #1794: 11.1 Residuals vs. Time plot: no refresh
@Yuri05 commented on [Fri Nov 04 2022](https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2401)

If I change the settings on the observed data tab in a simulation (scaling or weight): Residual vs. time Plots are not refreshed.
I have to close and reopen the simulation to see the changes.

And if I clear the mapping of observed data set: it is not removed from the plot even after reopening the simulation, so I need to create a new plot (this also applies to Predicted vs Observed plots)

---

@msevestre commented on [Mon Nov 07 2022](https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2401#issuecomment-1305623721)

Those are two issues here that I think we should split. 
/1 is not too much of a problem. It's a small bug but why not
/2 needst to be fixed for sure

---

@PavelBal commented on [Mon Nov 14 2022](https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2401#issuecomment-1313909430)

tbd, but if implementing 2., the residuals error must be updated, too.

---

@georgeDaskalakis commented on [Tue Nov 15 2022](https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2401#issuecomment-1315563032)

accidentally closed. reopening.

### PR DESCRIPTION
**Pull request by rwmcintosh**

### CODE REVIEW

File: src/OSPSuite.Core/Events/AnalysableEvents.cs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ﻿using System.Collections.Generic;
 using OSPSuite.Core.Domain;
 using OSPSuite.Core.Domain.Data;
+using OSPSuite.SimModel;
```
Comment (msevestre) [inline src/OSPSuite.Core/Events/AnalysableEvents.cs:4]: Why was this added? if htis is required, this is PROBABLY NOT The right ISimulation here..

**Review by msevestre — CHANGES_REQUESTED** [review]

File: src/OSPSuite.Presentation/Presenters/SimulationVsObservedDataChartPresenter.cs [hunk 2]
```diff
@@ -85,5 +87,11 @@ protected void AddUsedObservedDataToChart()
       }
 
       protected abstract void AddRunResultToChart();
+      
+      public void Handle(SimulationOutputMappingsChangedEvent eventToHandle)
+      {
+         if (eventToHandle.Simulation == _simulation)
```
Comment (msevestre) [inline src/OSPSuite.Presentation/Presenters/SimulationVsObservedDataChartPresenter.cs:93]: this can cras if Simulation is null (I think). so we usually do (Equals()) .
I may be wrong

File: src/OSPSuite.Core/Events/AnalysableEvents.cs (hunk 1 above)
Comment (rwmcintosh) [reply src/OSPSuite.Core/Events/AnalysableEvents.cs:4]: It got added at one point, but it's not used

**Review by msevestre — APPROVED** [review]

File: src/OSPSuite.Presentation/Presenters/SimulationVsObservedDataChartPresenter.cs (hunk 2 above)
Comment (rwmcintosh) [reply src/OSPSuite.Presentation/Presenters/SimulationVsObservedDataChartPresenter.cs:93]: I'll change it to Equals, but I don't think it should crash. if one is null

Comment (msevestre) [reply src/OSPSuite.Presentation/Presenters/SimulationVsObservedDataChartPresenter.cs:93]: Ok . Maybe old behavior then
